addons/account/models/mail_scheduled_message.py

```diff
@@ -0,0 +1,176 @@
+
+    @api.model
+    def _sending_settings_whitelist(self):
+        """ Sending Settings that can be used when generating and sending invoices.
+        """
+        return {
+            'sending_methods',
+            'invoice_edi_format',
+            'extra_edis',
+            'pdf_report',
+            'author_user_id',
+            'author_partner_id',
+            'mail_template',
+            'mail_lang',
+            'mail_body',
+            'mail_subject',
+            'mail_partner_ids',
+            'mail_attachments_widget',
+            'allow_fallback_pdf',
+        }
+
+    @api.model
+    def _post_messages_cron(self, limit=50):
+        # OVERRIDE
+        # To set OdooBot as a author of the messages from account.move.send.wizard
+        domain = [('scheduled_date', '<=', fields.Datetime.now())]
+        messages = self.search(domain, limit=limit)
+        messages_from_move = messages.filtered(lambda m: json.loads(m.notification_parameters or '{}').get('from_move_send_wizard'))
+        if messages_from_move:
+            _logger.info("Posting %s scheduled messages", len(messages_from_move))
+            messages_from_move.with_context(mail_notify_force_send=True).with_user(SUPERUSER_ID)._post_message(raise_exception=False)
+
+        super(MailScheduledMessage, messages - messages_from_move)._post_messages_cron(limit=limit)
+
+    def get_sending_method_and_edi_data(self):
+        """ This method used to determine if there any messages already scheduled to send
+            with EDI or sending methods other than 'email' for Invoice
+            :return: A dict with the following keys:
+                - sending_methods: A set of sending methods used in scheduled messages
+                - extra_edis:      A set of EDI methods used in scheduled messages
+        """
+        payloads = []
+        sending_methods = set()
+        extra_edis = set()
+
+        for scheduled_message in self:
+            if scheduled_send_payload := json.loads(scheduled_message.notification_parameters or '{}').get('scheduled_send_payload'):
+                payloads.append(scheduled_send_payload)
+
+        for payload in payloads:
+            # Exclude 'email' as it is necessary to allow scheduling multiple emails at once
+            sending_methods.update(sending_method for sending_method in payload.get('sending_methods', []) if sending_method != 'email')
+            extra_edis.update(payload.get('extra_edis', []))
+
+        return {
+            'sending_methods': sending_methods,
+            'extra_edis': extra_edis,
+        }
+
+    def post_log_note_at_unlink(self):
+        """ Post a log note on the move when the scheduled messages with "EDI or by Post" methods are cancelled by user. """
+        for scheduled_message in self:
+            if scheduled_message.is_note and \
+                    json.loads(scheduled_message.notification_parameters or '{}').get('from_move_send_wizard'):
```

Comment:
Use filtered, allows to gain one indent level directly :) also check we are on account.move model
```suggestion
        for scheduled_message in self.filtered(lambda s: (
            s.is_note and s.model == 'account.move' and
            json.loads(s.notification_parameters or '{}').get('from_move_send_wizard')
        )):
```